src/VisualStudio/Core/Def/Implementation/FindReferences/FindUsagesValueUsageInfoColumnDefinition.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.FindUsages
+{
+    /// <summary>
+    /// Custom column to display the reference kind/usage info for the Find All References window.
+    /// </summary>
+    [Export(typeof(ITableColumnDefinition))]
+    [Name(ColumnName)]
+    internal sealed class FindUsagesValueUsageInfoColumnDefinition : AbstractFindUsagesCustomColumnDefinition
```

Comment:
@olegtk so this will not load roslyn when like C++ do find all references? right?

Reply:
Definitely need to figure this out as well.

Reply:
The column doesn't seem to show unless explicitly added to the table control. MEF-exporting just seems to add it to the table column definition manager.

---
In reply to: [222372310](https://github.com/dotnet/roslyn/pull/30155#discussion_r222372310) [](ancestors = 222372310)

Reply:
It's still not clear to me though when it is MEF imported.  If it is imported even when C#/VB haven't been loaded... then that's not good :-/

Reply:
Ah, get your point. Probably @olegtk can answer that. Note that our approach is same as what C++ itself has been using for it's custom read/write column, which has been in place for at least a release.

---
In reply to: [222384111](https://github.com/dotnet/roslyn/pull/30155#discussion_r222384111) [](ancestors = 222384111)